pkg/queue/manager.go

```diff
@@ -300,13 +300,14 @@ func (m *Manager) RequeueWorkload(ctx context.Context, info *workload.Info, imme
 		return false
 	}
 
-	q.AddIfNotPresent(info)
+	newInfo := workload.NewInfo(&w)
```

Comment:
Can you explain why you think this solves the problem?

If there was a change in the object, the event would update the object, and `AddIfNotPresent` would prevent it to be added again.

Reply:
Refer to the log, we are always failing with update operation, this must be the reason that we stored the old version workload. So we should ensure to store the latest workload. The reason might be that the failed api update with the old version  requeue earlier than the newest one, then we stored the old ones. We can also find some clues from the log:
<img width="1910" alt="image" src="https://user-images.githubusercontent.com/18364341/166482500-5182d59c-fd98-4385-b5c6-519a945a5168.png">

Reply:
But that screenshot suggests that it wasn't actually added. So I'm not sure if this solves the issue or maybe it obscures a different one.

What's the link for this failure? I want to look at the entire logs.

Reply:
Here's the link https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/227/pull-kueue-test-integration-main/1520594730851766272/build-log.txt

Reply:
<img width="1919" alt="image" src="https://user-images.githubusercontent.com/18364341/166486355-6ae48b0b-483a-4260-9617-eee1e35021e2.png">
From the context, we know that the `assume-1` is return from the old version, it requeued successfully, the `assume-2` is from the new version, it requeued failed for old workload has already requeued.